x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/rollup/action/RollupShardStatus.java

```diff
@@ -49,16 +62,38 @@ public class RollupShardStatus implements Task.Status {
                 (Long) args[2],
                 (Long) args[3],
                 (Long) args[4],
-                (Long) args[5]
+                (Long) args[5],
+                (Long) args[6],
+                (Long) args[7],
+                (Long) args[8],
+                (Long) args[9],
+                (Long) args[10],
+                (RollupBeforeBulkInfo) args[11],
+                (RollupAfterBulkInfo) args[12]
             )
         );
 
-        PARSER.declareString(constructorArg(), SHARD_FIELD);
-        PARSER.declareString(constructorArg(), START_TIME_FIELD);
-        PARSER.declareLong(constructorArg(), IN_NUM_DOCS_RECEIVED_FIELD);
-        PARSER.declareLong(constructorArg(), OUT_NUM_DOCS_SENT_FIELD);
-        PARSER.declareLong(constructorArg(), OUT_NUM_DOCS_INDEXED_FIELD);
-        PARSER.declareLong(constructorArg(), OUT_NUM_DOCS_FAILED_FIELD);
+        PARSER.declareString(ConstructingObjectParser.constructorArg(), SHARD_FIELD);
+        PARSER.declareString(ConstructingObjectParser.constructorArg(), START_TIME_FIELD);
+        PARSER.declareLong(ConstructingObjectParser.constructorArg(), IN_NUM_DOCS_RECEIVED_FIELD);
+        PARSER.declareLong(ConstructingObjectParser.constructorArg(), OUT_NUM_DOCS_SENT_FIELD);
+        PARSER.declareLong(ConstructingObjectParser.constructorArg(), OUT_NUM_DOCS_INDEXED_FIELD);
+        PARSER.declareLong(ConstructingObjectParser.constructorArg(), OUT_NUM_DOCS_FAILED_FIELD);
+        PARSER.declareLong(ConstructingObjectParser.constructorArg(), TOTAL_DOC_COUNT);
+        PARSER.declareLong(ConstructingObjectParser.constructorArg(), TOTAL_SHARD_DOC_COUNT);
+        PARSER.declareLong(ConstructingObjectParser.constructorArg(), LAST_SOURCE_TIMESTAMP);
+        PARSER.declareLong(ConstructingObjectParser.constructorArg(), LAST_TARGET_TIMESTAMP);
+        PARSER.declareLong(ConstructingObjectParser.constructorArg(), LAST_INDEXING_TIMESTAMP);
+        PARSER.declareObject(
```

Comment:
@martijnvg this is the part that is not clear to me...how do we correctly parse "complex" objects.
I see there are multiple methods to do that but I am not sure which one is the correct one.

Reply:
Which  methods are you referring too? I think `declareObject(...)` is correct in this case. There is 1-1 relation for `RollupShardStatus` to RollupBeforeBulkInfo and RollupAfterBulkInfo, so no need to declare object array. We know what we need to parse (either `RollupAfterBulkInfo` or `RollupBeforeBulkInfo` class), so there is no need to use `declareNamedObject(...)`.

Reply:
So named objects is used when we need some logic to determine which actual object/class to parse. Probably especially if we are dealing with interfaces and multiple different implementations we need that (not in this case).